### PR TITLE
add DIP-10

### PR DESCRIPTION
**Pull request by gustavoguimaraes**
https://dynamicsetdollar.medium.com/dsd-v2-final-specs-voting-d17db9c8b5a

Voting is live:
https://dsd.finance/app/#/governance/candidate/0x6b358B24d1E209EdbE1DF36755B22c4A80549f7b

### CODE REVIEW

File: contracts/dao/Getters.sol [hunk 1]
```diff
@@ -95,10 +95,41 @@ contract Getters is State {
...
+    function globalInterestMultiplier() public view returns (uint256) {
+        return _state10.globalInterestMultiplier;
+    }
+
+    function expansionStartEpoch() public view returns (uint256) {
+        return _state10.expansionStartEpoch;
+    }
+
+    function dip10TotalRedeemed() public view returns (uint256) {
+        return _state10.dip10TotalRedeemable;
```
Comment (leruaa) [inline contracts/dao/Getters.sol]: Shouldn't be `_state10.dip10TotalRedeemed`?

File: contracts/dao/Getters.sol [hunk 2]
```diff
@@ -146,6 +177,85 @@ contract Getters is State {
...
+    function totalCDSDDeposited() public view returns (uint256) {
+        return _state10.totalCDSDDeposited;
+    }
+
+    function totalCDSDEarnable() public view returns (uint256) {
+        return _state10.totalCDSDEarnable;
+    }
+
+    function totalCDSDEarned() public view returns (uint256) {
+        return _state10.totalCDSDRedeemed;
```
Comment (leruaa) [inline contracts/dao/Getters.sol]: `_state10.totalCDSDEarned`?

File: contracts/dao/Getters.sol [hunk 3]
```diff
@@ -146,6 +177,85 @@ contract Getters is State {
...
+    function totalCDSDRedeemed() public view returns (uint256) {
+        return _state10.totalCDSDRedeemed;
+    }
+
+
+    function maxCDSDOutstanding() public view returns (uint256) {
+        return totalCDSDDeposited()
+            .add(totalCDSDEarnable())
+            .sub(totalCDSDEarned())
+            .sub(totalCDSDRedeemed());
```
Comment (acatinon) [inline contracts/dao/Getters.sol:254]: It seems this function always returns 0, Am I missing something?

File: contracts/dao/Getters.sol (hunk 1 above)
Comment (gustavoguimaraes) [reply contracts/dao/Getters.sol]: Good point! @devnulldsd , it was added in [this commit](https://github.com/dynamicsetdollar/dsd-hardhat/pull/1/commits/ba6c27750edb0a08e213ed7def90942fa343d0f4), any reasoning for this?

Comment (gustavoguimaraes) [reply contracts/dao/Getters.sol]: It looks incorrect. Thanks for pointing out @acatinon.

File: contracts/dao/Getters.sol (hunk 2 above)
Comment (gustavoguimaraes) [reply contracts/dao/Getters.sol]: May have been looked over. Thanks for pointing it out.

File: contracts/dao/Getters.sol (hunk 3 above)
Comment (devnulldsd) [reply contracts/dao/Getters.sol]: Please check the last commits for changes to the global accounting, I think its correct now, with the recent changes

**Pull request merged**